stubs/psycopg2/psycopg2/__init__.pyi

```diff
@@ -27,19 +27,28 @@ from psycopg2._psycopg import (
     Warning as Warning,
     __libpq_version__ as __libpq_version__,
     apilevel as apilevel,
-    connection as connection,
-    cursor as cursor,
+    connection,
+    cursor,
     paramstyle as paramstyle,
     threadsafety as threadsafety,
 )
 
 _T_conn = TypeVar("_T_conn", bound=connection)
 
 @overload
-def connect(dsn: str, connection_factory: Callable[..., _T_conn], cursor_factory: None = None, **kwargs: Any) -> _T_conn: ...
+def connect(
+    dsn: str | None,
+    connection_factory: Callable[..., _T_conn],
+    cursor_factory: Callable[..., cursor] | None = None,
+    **kwargs: Any,
+) -> _T_conn: ...
 @overload
 def connect(
-    dsn: str | None = None, *, connection_factory: Callable[..., _T_conn], cursor_factory: None = None, **kwargs: Any
+    dsn: str | None = None,
+    *,
+    connection_factory: Callable[..., _T_conn],
+    cursor_factory: Callable[..., cursor] | None = None,
+    **kwargs: Any,
```

Comment:
These two overloads look identical to me now.

Reply:
Mmm. I was deleting the first overload then remembered why I kept both.
In the first, the function can be called with positional arguments while in the second only `dsn` can be positional (because we need to give it a default value and can't give a default to `connection_factory`) so this test
`assert_type(psycopg2.connect("test-conn", MyConnection), MyConnection)` fails. I added this test case so this doesn't regress.

Reply:
Good point, I missed that.